flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BufferOrEvent.java

```diff
@@ -43,26 +44,32 @@
 
 	private int channelIndex;
 
-	BufferOrEvent(Buffer buffer, int channelIndex, boolean moreAvailable) {
+	private final int size;
+
+	public BufferOrEvent(Buffer buffer, int channelIndex, boolean moreAvailable, int size) {
```

Comment:
why are we passing this new field explicitly? Why not
```
this.size = buffer.getSize();
```
?

Further more, why do we need `size` field at all? Why not implement:
```
int getSize() {
  return buffer == null ? 0 : buffer.getSize();
}
```

Reply:
For event buffer, we might not return 0, so we have the explicit size field, but we do not need pass the size for buffer constructor as you suggested.